### PR TITLE
feat(issue-171): Handle errors more efficiently

### PR DESCRIPTION
**Pull request by jisung-lee7**
## Issue Link

[issue-171](https://github.com/kafelix496/coding-for-fun/issues/171)

## Notes

-

### CODE REVIEW

File: src/utils/http-errors.ts
Comment (jiyeol-lee) [inline src/utils/http-errors.ts:18]: I rather change it to object instead of 4 parameters like the following code
```typescript
export const createError = (error: { title: string, description: string, statusCode: number }) => {
  const { title, description, statusCode } = error

 // blah blah
}
```

File: src/utils/http-errors.ts
Comment (jiyeol-lee) [inline src/utils/http-errors.ts:17]: I don't think we need `message` field.

File: src/app/api/github/repos/[owner]/[repo]/pulls/[pullNumber]/reviews/route.ts
Comment (jiyeol-lee) [inline src/app/api/github/repos/[owner]/[repo]/pulls/[pullNumber]/reviews/route.ts:64]: Are you 100% sure `error.name`, `error.response.data.message`, `error.response.data.status`, and `error.response.data.documentation_url` are all valid data?

**Comment by jisung-lee7** [comment]
please don't merge yet.

File: src/utils/http-errors.ts
Comment (jiyeol-lee) [inline src/utils/http-errors.ts:1]: Try to write a test code for this one.

File: src/utils/http-errors.test.ts
Comment (jiyeol-lee) [inline src/utils/http-errors.test.ts:42]: use "'" instead of "`"

File: src/utils/http-errors.test.ts
Comment (jiyeol-lee) [inline src/utils/http-errors.test.ts:59]: use "'" instead of "`"

File: src/utils/http-errors.test.ts
Comment (jiyeol-lee) [inline src/utils/http-errors.test.ts:8]: when "what" not provided?

File: src/types/server.ts
Comment (jiyeol-lee) [inline src/types/server.ts:7]: Change this to `description`

**Review by jiyeol-lee — CHANGES_REQUESTED** [review]

File: src/utils/http-errors.ts
Comment (jiyeol-lee) [inline src/utils/http-errors.ts:56]: can you add test for this use-case?

Comment (jisung-lee7) [reply src/utils/http-errors.ts]: sure why not ! 😎

File: src/utils/http-errors.ts
Comment (jiyeol-lee) [inline src/utils/http-errors.ts:53]: Remove console before you merge

**Review by jiyeol-lee — APPROVED** [review]

File: src/utils/http-errors.ts
Comment (jisung-lee7) [reply src/utils/http-errors.ts:53]: 하 ㅋㅋㅋㅋㅋㅋㅋㅋ쏘리 부끄러운 디버깅현장을 목격당했구만?

**Pull request merged**
